tests/components/lcn/test_services.py

```diff
@@ -0,0 +1,399 @@
+    await setup_component(hass)
+
+    # success
+    with patch.object(
+        MockModuleConnection, "lock_keys_tab_a_temporary"
+    ) as lock_keys_tab_a_temporary:
+        await hass.services.async_call(
+            DOMAIN,
+            "lock_keys",
+            {
+                "address": "pchk.s0.m7",
+                "state": "0011TT--",
+                "time": 10,
+                "time_unit": "s",
+            },
+            blocking=True,
+        )
+
+    assert lock_keys_tab_a_temporary.await_args.args == snapshot(
+        name="lock_keys_tab_a_temporary"
+    )
+
+    # wrong table
+    with (
+        patch.object(
+            MockModuleConnection, "lock_keys_tab_a_temporary"
+        ) as lock_keys_tab_a_temporary,
+        pytest.raises(ValueError),
+    ):
+        await hass.services.async_call(
+            DOMAIN,
+            "lock_keys",
+            {
+                "address": "pchk.s0.m7",
+                "table": "b",
+                "state": "0011TT--",
+                "time": 10,
+                "time_unit": "s",
+            },
+            blocking=True,
+        )
+
+
+@patch("pypck.connection.PchkConnectionManager", MockPchkConnectionManager)
+async def test_service_dyn_text(
+    hass: HomeAssistant, snapshot: SnapshotAssertion
+) -> None:
+    """Test dyn_text service."""
+    await async_setup_component(hass, "persistent_notification", {})
+    await setup_component(hass)
+
+    with patch.object(MockModuleConnection, "dyn_text") as dyn_text:
+        await hass.services.async_call(
+            DOMAIN,
+            "dyn_text",
+            {"address": "pchk.s0.m7", "row": 1, "text": "text in row 1"},
+            blocking=True,
+        )
+
+    assert dyn_text.await_args.args == snapshot(name="dyn_text")
+
+
+@patch("pypck.connection.PchkConnectionManager", MockPchkConnectionManager)
+async def test_service_pck(hass: HomeAssistant, snapshot: SnapshotAssertion) -> None:
+    """Test pck service."""
+    await async_setup_component(hass, "persistent_notification", {})
+    await setup_component(hass)
+
+    with patch.object(MockModuleConnection, "pck") as pck:
+        await hass.services.async_call(
+            DOMAIN,
+            "pck",
+            {"address": "pchk.s0.m7", "pck": "PIN4"},
+            blocking=True,
+        )
+
+    assert pck.await_args.args == snapshot(name="pck")
+
+
+@patch("pypck.connection.PchkConnectionManager", MockPchkConnectionManager)
+async def test_service_called_with_invalid_host_id(
+    hass: HomeAssistant, snapshot: SnapshotAssertion
+) -> None:
+    """Test service was called with non existing host id."""
+    await async_setup_component(hass, "persistent_notification", {})
+    await setup_component(hass)
+
+    with patch.object(MockModuleConnection, "pck") as pck, pytest.raises(ValueError):
+        await hass.services.async_call(
+            DOMAIN,
+            "pck",
+            {"address": "foobar.s0.m7", "pck": "PIN4"},
```

Comment:
Can we use constants for the parameters?
So only `pck` and `address`? And for the other tests as well

Reply:
Short question: Do you mean constants for the keys (e.g., `"address"`) or for the values (e.g., `"foobar.s0.m7"`)?

Reply:
The keys and the service names

Reply:
Done